UX_PRINCIPLES.md

```diff
@@ -0,0 +1,79 @@
+### Embrace Design Constraints
+- Use characters (icons), progress bars, and color in your output
+- Provide an option to remove characters, progress bars and color (can break automated build tasks, for example, that humans aren’t reading/watching anyway)
+### Generate Humane Output
+- Make output human readable and concise by default. Offer more/less verbose output as an option via a flag.
+- Examine the current state of the system and make output contextual
+- Identify for analogs and antilogs. Do repeat CLI experiences that you’ve enjoyed elsewhere; Don’t repeat mistakes others have made. (e.g. In Habitat, building a package follows a format like bundle install)
+- Write output with a clear messages that a) sets the stage with “what we’re about to do”, b) progress “what is happening now”, and c) “did it fail or succeed?” and if it failed explain why, then provide a recommendation/next step. If it succeeds, celebrate it with memorable/playful message.
+### Expose Your Personality
+- It’s a machine, but it doesn’t have to feel like one. Remind your users that there are people behind this product and that they care about your experience.
+- Don’t overuse this principle by being overly silly or ever-present (that will be annoying), but a couple of inside jokes or playfully-worded sentences (especially during moments of success) sprinkled throughout keeps things interesting and personable (e.g. when a plan builds successfully out last line of output is “I love it when a plan.sh comes together.”)
+### Provide Feedback
+- Provide a sense of how long something will take if it’s going to longer than a few seconds with progress bars and/or honest message like “this may take several minutes” and show that it is progressing (not stuck)
+### Avoid Dead-ends
+- Error message format / help links and instructions
+
+## Instrumentation
+### Measure Effectiveness
+- It’s the command line so you can’t get any analytics, right? Wrong. As with a web or mobile application, you can still implement tracking with common tools like Google Analytics. That said, you have some unique challenges to address:
+ - Treat the terminal experience as a more private and sacred interface for your users. Unlike the app world where there is obvious reliance upon the internet and thus an implied notion of less privacy, the command line experience feels more confined to the walls of your machine.
+ - We’ve seen examples of community backlash when analytics were opted in by default (see homebrew) and it was not pretty. Given this more private setting, be courteous and transparent to retain trust and get the data you need. Some tips on accomplishing include:
+  - Set analytics as opt out by default unless you are absolutely certain (and have communicated thoroughly elsewhere) that you’re tracking things.
+  - Be explicit on what/when/why/how your sending data. Put it in your permission request test, code comments, and docs. Be transparent; have nothing to hide.
```

Comment:
`you're` is a contraction. "Put it in _you are_ permission request test" doesn't sound quite right.